attiny402_firmware/src/led.cpp

```diff
@@ -0,0 +1,52 @@
+#include "led.h"
+
+WS2812 LED(1);                  //rgb led controller instance
+cRGB val;                       //used for global color changes
+uint8_t a_rgb[3] = {0,0,255};   //used for automatic color changes
+uint8_t rgb_cycle_state = 0;    //used to keep track of the colors being changed in the rgb cycler
+
+ISR(TCA0_CMP0_vect) {
+    TCA0.SINGLE.INTFLAGS = TCA_SINGLE_CMP0_bm;          //clear interrupt flag
+
+    switch(led_mode){
+        case 0:
+            return;                                     //in led_mode 0 just return so you can set the color externally
+        case 1:
+            a_rgb[rgb_cycle_state]+=2;                  //no questions. this cycles smoothly through rgb colors and it just works
+            a_rgb[(rgb_cycle_state+1) % 3]-=2;
+            if(a_rgb[rgb_cycle_state] > 253) rgb_cycle_state++;
+            if(rgb_cycle_state > 2) rgb_cycle_state = 0;
+            set_led_color(a_rgb[0], a_rgb[1], a_rgb[2]);
+        case 2:
```

Comment:
Leave a comment explaining why this case is empty or remove it

Reply:
fixed